### PR TITLE
MULE-11138: Make easier to work with UDT on DB connector

### PR DESCRIPTION
**Pull request by pablokraan**
_ Adding support to create java.sql.Struct and java.sql.Array values from Object[] and List values
_ Adding EL functions to create array and struct values from MEL expressions

### CODE REVIEW

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        super(delegate, transactionalAction, connectionReleaseListener, paramTypeResolverFactory);
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException
+    {
+        Method createArrayMethod;
+        try
+        {
+            createArrayMethod = delegate.getClass().getMethod("createARRAY", String.class, Object.class);
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java]: don't do this lookup every time. cache this method on the constructor and wrap it in some kind of strategy object

File: modules/db/src/main/java/org/mule/module/db/internal/domain/type/ArrayResolvedDbType.java [hunk 2]
```diff
@@ -32,9 +37,35 @@ public ArrayResolvedDbType(int id, String name)
...
     }
 
+    /**
+     * Creates error message for the case when a given class is not supported
+     *
+     * @param value value that was attempted to be converted
+     * @return the error message for the provided value's class
+     */
+    protected static String createUnsupportedTypeErrorMessage(Object value) {
+        return format("Cannot create a %s from a value of type %s", Struct.class.getName(), value.getClass());
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/type/ArrayResolvedDbType.java]: Shouldn't you use the class simpleName instead?

File: modules/db/src/main/java/org/mule/module/db/internal/el/AbstractDbFunction.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import org.mule.module.db.internal.domain.transaction.TransactionalAction;
+import org.mule.module.db.internal.resolver.database.DbConfigResolver;
+
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Provides a base class for DB functions that creates custom DB types.
+ */
+public abstract class AbstractDbFunction  implements ExpressionLanguageFunction
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/el/AbstractDbFunction.java]: good luck with this in Mule 4. I wish you all the best

File: modules/db/src/test/java/org/mule/module/db/integration/function/DbCreateArrayFunctionTestCase.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        testDatabase.createStoredProcedureUpdateZipCodes(dataSource);
+        testDatabase.createStoredProcedureUpdateContactDetails(dataSource);
+    }
+
+    @Test
+    public void createsDefaultTypeArray() throws Exception
+    {
+        LocalMuleClient client = muleContext.getClient();
+
+        MuleMessage response = client.send("vm://createsDefaultTypeArray", NORTHWEST.getZips(), null);
```
Comment (marianogonzalez) [inline modules/db/src/test/java/org/mule/module/db/integration/function/DbCreateArrayFunctionTestCase.java]: do you really need to use vm for this tests? CAn't you just runFlow()?

File: modules/db/src/main/java/org/mule/module/db/internal/domain/type/ArrayResolvedDbType.java (hunk 2 above)
Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/type/ArrayResolvedDbType.java]: No, full qualified name is required to univocally reference a class

File: modules/db/src/main/java/org/mule/module/db/internal/el/AbstractDbFunction.java (hunk 3 above)
Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/el/AbstractDbFunction.java]: I don't need good luck, I'm Muleman

File: modules/db/src/test/java/org/mule/module/db/integration/function/DbCreateArrayFunctionTestCase.java (hunk 4 above)
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/function/DbCreateArrayFunctionTestCase.java]: No really required, I did it in this way as all the tests in the module use the same approach. I can change it

**Pull request merged**